### PR TITLE
Created an email view to send email to the event participants

### PR DESCRIPTION
**Pull request by nikhilgupta1211**
Added a link to email in event#show and created a method in events_helper to retrieve user emails based on state of the travel request

### CODE REVIEW

File: app/helpers/events_helper.rb [hunk 1]
```diff
@@ -10,5 +10,13 @@ def event_title_line(event)
     t(:event_title_line, :name => event.name, :start => l(event.start_date),
       :end => l(event.end_date), :country => event.country_code)
   end
-end
 
+  def users_for_event(state)
+  	if state != 'all'
```
Comment (bgeuken) [inline app/helpers/events_helper.rb]: Please use 2 whitespaces for indentation, and no tabs.

We don't follow a particular style guide right now, but [this one](https://github.com/bbatsov/ruby-style-guide) is widely used in ruby world and might be a good reference.

Comment (nikhilgupta1211) [inline app/helpers/events_helper.rb]: ok :)

File: app/helpers/events_helper.rb [hunk 2]
```diff
@@ -10,5 +10,13 @@ def event_title_line(event)
...
+    	req = @event.requests.select{|e| e.state == state}
```
Comment (bgeuken) [inline app/helpers/events_helper.rb]: It would be better to use a db query here, so ```  @event.requests.where(...) ```

Comment (nikhilgupta1211) [inline app/helpers/events_helper.rb]: Done

File: app/views/events/email.html.haml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+%h2 Email the participants of #{@event.name}
+%br
+.span10
+  .dropdown
+    %a.dropdown-toggle{"data-toggle" => "dropdown", :href => "#"} Select recipents
```
Comment (bgeuken) [inline app/views/events/email.html.haml]: That should be Select recipients

File: app/helpers/events_helper.rb [hunk 4]
```diff
@@ -10,5 +10,13 @@ def event_title_line(event)
...
       :end => l(event.end_date), :country => event.country_code)
   end
-end
 
+  def users_for_event(state)
+    if state != 'all'
+      req = @event.requests.where(state: state)
+    else
+      req = @event.requests
+    end
```
Comment (bgeuken) [inline app/helpers/events_helper.rb]: btw you could shorten this:
```ruby
req = @event.requests
req = req.where(state: state) if state != 'all'
```

Comment (nikhilgupta1211) [inline app/helpers/events_helper.rb]: ok :)

File: spec/features/events_spec.rb [hunk 5]
```diff
@@ -52,4 +52,33 @@
...
+    page.should have_field('To', with: 'gial.ackbar@rebel-alliance.org,luke.skywalker@rebel-alliance.org,evram.lajaie@rebel-alliance.org,c3po@droids.com')
+    click_link "Select recipients"
+    find('a', :text => "Approved").click
+    page.should have_field('To', with: '')
+    click_link "Select recipients"
+    find('a', :text => "Accepted").click
+    page.should have_field('To', with: '')
+
+    click_link "Cancel"
+    current_path.should == event_path('160780002') # 160780002 is the id of Death Star Event
```
Comment (bgeuken) [inline spec/features/events_spec.rb]: This should work as well: ```ruby event_path(events(:party)) ```

And then you won't need the comment.

Comment (nikhilgupta1211) [inline spec/features/events_spec.rb]: Ok

File: spec/features/events_spec.rb [hunk 6]
```diff
@@ -52,4 +52,33 @@
...
+    click_link "Select recipients"
+    find('a', :text => "Approved").click
+    page.should have_field('To', with: '')
+    click_link "Select recipients"
+    find('a', :text => "Accepted").click
+    page.should have_field('To', with: '')
+
+    click_link "Cancel"
+    current_path.should == event_path(events(:party))
+  end
```
Comment (bgeuken) [inline spec/features/events_spec.rb]: One thing I forgot... We should add a test that ensures that a user who has not the required permissions can not see the email link on the event page. And the same for users that are not logged in.

Comment (nikhilgupta1211) [inline spec/features/events_spec.rb]: Done :)

**Comment by nikhilgupta1211** [comment]
I have updated the email view with Bootstrap 3.

**Comment by bgeuken** [comment]
@nikhilgupta1211 Sorry for the messup. I accidentally merged PR#47 when I wanted to merge this one. I reverted it now.

**Pull request merged**